site/content/integrate/apps/structure/forms/_index.md

```diff
@@ -0,0 +1,712 @@
+            },
+            {
+                "name": "channelid",
+                "label": "channelid",
+                "type": "text",
+                "subtype": "input",
+                "description": "The ID of the channel"
+            }
+        ],
+        "submit": {
+            "path": "/sub",
+        }
+    }
+}
+```
+{{</collapse>}}
+{{<collapse id="slash-command-positional-example" title="Example form with positional arguments">}}
+```json
+{
+    "location": "sub",
+    "label": "sub",
+    "description": "Subscribe to an event",
+    "form": {
+        "title": "Subscribe to an event",
+        "header": "Subscribe to a Mattermost Server event",
+        "icon": "icon.png",
+        "fields": [
+            {
+                "name": "eventname",
+                "label": "eventname",
+                "type": "text",
+                "subtype": "input",
+                "description": "The name of the event to subscribe to",
+                "is_required": true,
+                "position": 1
+            },
+            {
+                "name": "teamid",
+                "label": "teamid",
+                "type": "text",
+                "subtype": "input",
+                "description": "The ID of the team",
+                "position": 2
+            },
+            {
+                "name": "channelid",
+                "label": "channelid",
+                "type": "text",
+                "subtype": "input",
+                "description": "The ID of the channel",
+                "position": 3
+            }
+        ],
+        "submit": {
+            "path": "/sub",
+        }
+    }
+}
+```
+{{</collapse>}}
+
+During [autocomplete]({{<ref "/integrate/slash-commands">}}), the user can open the form in a modal dialog to finish entering command arguments.
+Any fields not supported by commands, such as markdown fields, or form attributes not visible in commands, such as the title, will be shown when the form is opened as a modal dialog.
+
+![Screenshot of slash command autocomplete](slash-command-autocomplete.png)
+
+![Screenshot of autocomplete modal dialog](slash-command-autocomplete-modal-dialog.png)
+
+When the slash command is executed, the form's `submit` call will be performed. The call request will include the App, user, post, root post (if any), channel, and team IDs in the context.
+
+## End-to-end examples
+
+### Click a button in the channel header
+
+{{<collapse id="click_channel_header-client_submit_request" title="Client submit request">}}
+```http request
+POST /plugins/com.mattermost.apps/api/v1/call HTTP/1.1
+Content-Type: application/json
+
+{
+    "path":"/send",
+    "context":{
+        "app_id":"hello-world",
+        "location":"/channel_header/send-button",
+        "channel_id":"zot5mcxxcfbqddgif88sji6ziy",
+        "team_id":"143qpkqxfiy5xqqf33w51w8abe",
+        "track_as_submit":true,
+        "user_agent":"webapp"
+    },
+    "expand":{}
+}
```

Comment:
This payload is coming from the frontend and going to the App framework. Some devs may get confused and think that they need to be able to parse this exact payload here, when in reality they don't ever receive this. These were originally put in the docs when we were deciding what should be sent by the client, but this is more-or-less established now. 0/5 on the value it has to include these payloads in the docs

Reply:
Good point! I'll remove those sections.
They can be easily added back in (and generated) on demand.